itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/advanced/LogsTest.java

```diff
@@ -0,0 +1,169 @@
+		this.consoleView.open();		
+	}
+
+	@Test
+	public void shouldShowLogFromApplicationPodContextMenu() {
+		OpenShiftResource pod  = getApplicationPod(serviceName, getOpenShiftProject(requiredProject.getProjectName(), explorer));
+		String podName = pod.getName();
+		pod.select();
+		new ContextMenuItem(OpenShiftLabel.ContextMenu.POD_LOG).select();
+
+		new WaitUntil(new ConsoleHasSomeText(), TimePeriod.DEFAULT);
+		new WaitUntil(new ConsoleHasNoChange(TimePeriod.getCustom(WAIT_CONSOLE_NO_CHANGE)), TimePeriod.VERY_LONG);
+
+		assertTrue("Console label is incorrect, it should contains project name and pod name.\n" + "but label is: " + consoleView.getConsoleLabel(),
+				consoleView.getConsoleLabel().contains(requiredProject.getProjectName() + "\\" + podName));
+		assertTrue("Console text should contain output from EAP runtime",
+				consoleView.getConsoleText().contains("Admin console is not enabled"));
+	}
+
+	@Test
+	public void shouldShowLogFromBuildPodContextMenu() {
+		OpenShiftResource pod = getBuildPod(serviceName, getOpenShiftProject(requiredProject.getProjectName(), explorer));
+		String podName = pod.getName();
+		pod.select();
+		new ContextMenuItem(OpenShiftLabel.ContextMenu.BUILD_LOG).select();
+
+		new WaitUntil(new ConsoleHasSomeText(), TimePeriod.DEFAULT);
+		new WaitUntil(new ConsoleHasNoChange(TimePeriod.getCustom(WAIT_CONSOLE_NO_CHANGE)), TimePeriod.VERY_LONG);
+
+		assertTrue("Console label is incorrect, it should contain project name and name of build pod.\n"
+				+ "but label is: " + consoleView.getConsoleLabel(), 
+				consoleView.getConsoleLabel().contains(requiredProject.getProjectName() + "\\" + podName));
+		try {
+			new WaitUntil(new ConsoleHasText("Push successful"), 
+					TimePeriod.getCustom(WAIT_CONSOLE_PUSH_SUCCESS));
+		} catch (WaitTimeoutExpiredException ex) {
+			fail("There should be output of succesful build in console log, but there is not.\n"
+					+ "Check whether output has not changed. Assumed output in the end of log is 'Push successful'");
+		}
+	}
+
+	private OpenShiftProject getOpenShiftProject(String projectName, OpenShiftExplorerView explorer) {
+		OpenShift3Connection connection = explorer.getOpenShift3Connection(requiredConnection.getConnection());
+		OpenShiftProject project = connection.getProject(projectName);
+		assertThat("Could not find project " + projectName + " in OpenShift Explorer",  project, notNullValue());
+		project.expand();
+		return project;
+	}
+
+	private OpenShiftResource getBuildPod(String name, OpenShiftProject project) {
+		assertTrue(!StringUtils.isBlank(name));
+
+		for (OpenShiftResource pod : project.getOpenShiftResources(Resource.BUILD)) {
+			String podName = pod.getName();
+			if (!StringUtils.isBlank(podName)) {
+				if (podName.startsWith(name)) {
+					return pod;
+				}
+			}
+		}
+		fail("Build pod for service " + serviceName + " was not found");
+		return null;
+	}
+
+	private OpenShiftResource getApplicationPod(String name, OpenShiftProject project) {
```

Comment:
I'd guess that this should be moved to some utils class? It looks as if this has a similar purpose like  DeleteResourceTest#getApplicationPod (but is using core vs ui means like the latter). If using core classes is intentional/required I'd suggest adding this to OpenShift3NativeResourceUtils with parameters.